File: frontend/src/microservices/content/objects/CreateOrderCancellationMessageRequest.ts
export type CreateOrderCancellationMessageRequest = {
  threadId: string
  orderId: string
  reason: string
}

export const createOrderCancellationMessageRequestFromJson = (json: string): CreateOrderCancellationMessageRequest =>
  JSON.parse(json) as CreateOrderCancellationMessageRequest

export const createOrderCancellationMessageRequestToJson = (value: CreateOrderCancellationMessageRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/feedback/objects/CreateOrderCancellationMessageRequest.scala
// CreateOrderCancellationMessageRequest：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateOrderCancellationMessageRequest(threadId: String, orderId: String, reason: String)
object CreateOrderCancellationMessageRequest:
  given sourceEncoder: Encoder[CreateOrderCancellationMessageRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateOrderCancellationMessageRequest] = deriveDecoder
